### ISSUE TITLE
'alpha' kwarg overrides facecolor='none' when plotting circle

### ISSUE THREAD

**Issue by dstansby** [matplotlib/matplotlib#7478]
Minimal example:

```
import matplotlib.pyplot as plt

plt.figure()
ax = plt.subplot()
c = plt.Circle((0, 0), 1, facecolor='none', alpha=1)
ax.add_artist(c)
plt.show()
```
This plots a black filled in circle. I expected facecolor='none' to override alpha=1 and set a transparent facecolor, but instead it seems that alpha overrides the facecolor argument.

Present on 2.0.0b4+2839.g631e079 (installed from source using pip on OSX)

**Comment by tacaswell** [comment]
'none' maps to `(0, 0, 0, 0)` which is why it is black, not clear to me why this is happening :disappointed: 

attn @anntzer 

related to #7430

**Comment by anntzer** [comment]
to_rgba is implemented such that "none" always maps to (0, 0, 0, 0) regardless of the alpha value.  Unfortunately, currently the color is *first* converted from "none" to (0, 0, 0, 0) and *then* alpha=1 is applied, resulting in (0, 0, 0, 1) (i.e., the alpha-forcingness of "none" has been lost).

1.5.3 avoided this issue by storing `_original_facecolor` as "none", and applying alpha was (and is still) implemented as setting the alpha attribiute, then re-calling the facecolor and edgecolor setters with the previous value, so the "none" stored as `_original_facecolor` would override the alpha setting.

The issue bisects back to 05e4ed9a420441e1227d394bc7e107503d03667a (attn @efiring), so it is a regression.

**Comment by efiring** [comment]
I'm not going to try it right now, but I think the solution here *might* be to change `Patch.update_from` to directly copy the private underlying variables, as `Artist.update_from` does, instead of using the getters and setters.

**Comment by efiring** [comment]
Closed by #7504.
